packages/atomic/src/utils/accessibility-utils.tsx

```diff
@@ -17,7 +18,58 @@ export function AriaLiveRegion(regionName: string) {
 
   return (component: InitializableComponent, setterName: string) => {
     Object.defineProperty(component, setterName, {
-      set: (message) => dispatchMessage(message),
+      set: (message: string) => dispatchMessage(message),
     });
   };
 }
+
+export interface FocusTargetController {
+  setTarget(element: HTMLElement | undefined): void;
+  focusAfterSearch(): void;
+}
+
+export function FocusTarget() {
+  return (component: InitializableComponent, setterName: string) => {
+    const {componentWillLoad} = component;
+
+    component.componentWillLoad = function () {
+      componentWillLoad && componentWillLoad.call(this);
+      const {componentDidRender} = this;
+      let focusAfterSearch = false;
+      let lastSearchId: string | undefined = undefined;
+      let element: HTMLElement | undefined = undefined;
+
+      function tryFocusElement(bindings: Bindings) {
```

Comment:
This function is called at two different places. It's called in `setTarget` (bound to the `ref`) and `componentDidRender`.

The reason for that, is that `setTarget` is only executed when one particular element is re-rendered, while `componentDidRender` is executed after every search (or other state change). If an element already exists and we're waiting for a new search, then this needs to be called from `componentDidRender`. Inversely, I don't remember why, but when a new component is rendered it looks like we have to focus it from `setTarget`/`ref`.